extensions/pyRevitTools.extension/checks/cad_audit_check.py

```diff
@@ -0,0 +1,189 @@
+    flexform_comp.append(RadioButton("active_view", "in only the active view", False, GroupName="grp"))
+    flexform_comp.append(Separator())
+    flexform_comp.append(Button("CANCEL", on_click=ButtonClass.cancel_clicked))
+    flexform_comp.append(Button("OK"))
+
+    user_input = FlexForm("Audit of CAD instances in the current project", flexform_comp, Width=500, Height=200) # returns a FlexForm object
+    user_input.show()
+    user_input_dict = user_input.values # This is a dictionary
+    if not bool(user_input_dict): # check if dict not empty (a bool of non-empty dict is True)
+        script.exit()
+
+    return user_input_dict
+
+def get_load_stat(cad, is_link):
+    """ Loaded status from the import instance's CADLinkType """
+    cad_type = doc.GetElement(cad.GetTypeId()) # Retreive the type from the instance
+    
+    if is_link:
+        exfs = cad_type.GetExternalFileReference()
+        status = exfs.GetLinkedFileStatus().ToString()
+        if status == "NotFound":
+            status = ":cross_mark: NotFound"
+        elif status == "Unloaded":
+            status = ":heavy_multiplication_x: Unloaded"        
+    else:
+        status = ":warning: IMPORTED" # Not an external reference
+
+    return status
+
+
+def checkModel(doc, output):
+    timer = Timer()
+    output = script.get_output()
+    output.close_others()
+    output.set_title("CAD audit of model '{}'".format(doc.Title))
+    output.set_width (1700)
+    
+    coll_mode = get_user_input()["active_view"]
+    
+    table_data = [] # store array for table formatted output
+    row_head = ["No", "Select/Zoom", "DWG instance", "Loaded status", "Workplane or single view", "Duplicate", "Workset", "Creator user", "Location site name"] # output table first and last row
+    cad_instances = collect_cadinstances(coll_mode)
+    for cad in cad_instances:
+        count = 0
+        cad_id = cad.Id
+        cad_is_link = cad.IsLinked
+        cad_name = cad.Parameter[DB.BuiltInParameter.IMPORT_SYMBOL_NAME].AsString()
+
+        table_row = []
+        table_row.append(count+1)
+        table_row.append(output.linkify(cad_id, title="{}".format("Select")))
+        table_row.append(cad_name)
+        table_row.append(get_load_stat(cad, cad_is_link)) # loaded status
+
+        # if the instance has an owner view, it was placed on the active view only (bad, so give warning and show the view name)
+        # if the instance has no owner view, it should have a level or workplane (good)
+        cad_own_view_id = cad.OwnerViewId
+        if cad_own_view_id == DB.ElementId.InvalidElementId:
+            table_row.append(doc.GetElement(cad.LevelId).Name)
+        else:
+            cad_own_view_name = doc.GetElement(cad_own_view_id).Name
+            table_row.append(":warning: view '{}'".format(cad_own_view_name))
+        table_row.append(":warning:" if cad_name in [row[2] for row in table_data] else "-") # If the name is already in table_data, it is a duplicat (bad)
```

Comment:
It is better to manage a separate set with the names for quicker membership checks

```
seen_names = set()
# ... 
    table_row = [
        # ...
        ":warning:" if cad_name in seen_names else "-",
        # ...
    ]
    seen_names.add(cad_name)
```